Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -5778,21 +5781,21 @@ class BBQJIT {
         m_frameSizeLabel = m_jit.moveWithPatch(TrustedImmPtr(nullptr), m_scratchGPR);
 
         // Because we compile in a single pass, we always need to pessimistically check for stack underflow/overflow.
-        ScratchScope<1, 0> scratches(*this);
-        m_jit.subPtr(GPRInfo::callFrameRegister, m_scratchGPR, MacroAssembler::stackPointerRegister);
-        m_jit.move(MacroAssembler::stackPointerRegister, m_scratchGPR);
+        ASSERT(m_scratchGPR == GPRInfo::nonPreservedNonArgumentGPR0);
```

Comment:
Would `m_scratchGPR` ever be anything other than `GPRInfo::nonPreservedNonArgumentGPR0`?  If not, can we make `m_scratchGPR` a constexpr and change this into a `static_assert`?

Reply:
We would like to do it, but we need to decide the name, and I don't think this is what this patch should do. So right now, we are just keeping it.

Reply:
That's fair.  SGTM.